sklearn/tests/test_discriminant_analysis.py

```diff
@@ -540,36 +540,31 @@ def test_qda_regularization():
     # The default is reg_param=0. and will cause issues when there is a
     # constant variable.
 
-    # Fitting on data with constant variable triggers an UserWarning.
-    collinear_msg = "Variables are collinear"
+    # Fitting on data with constant variable without regularization
+    # triggers a LinAlgError.
+    msg = r"The covariance matrix of class .+ is not full rank"
     clf = QuadraticDiscriminantAnalysis()
-    with pytest.warns(UserWarning, match=collinear_msg):
+    with pytest.warns(linalg.LinAlgWarning, match=msg):
         y_pred = clf.fit(X2, y6)
 
-    # XXX: RuntimeWarning is also raised at predict time because of divisions
-    # by zero when the model is fit with a constant feature and without
-    # regularization: should this be considered a bug? Either by the fit-time
-    # message more informative, raising and exception instead of a warning in
-    # this case or somehow changing predict to avoid division by zero.
-    with pytest.warns(RuntimeWarning, match="divide by zero"):
-        y_pred = clf.predict(X2)
-    assert np.any(y_pred != y6)
```

Comment:
With the `LinAlgWarning`, is this `RuntimeWarning` still being raised?

Reply:
So it means that we need to record the warning and check that both warnings are raised or shall we catch the division by zero warning that is not informative then?

Reply:
I've put back the test, catching all the raised warnings.